source/beamline/SGM/monochromator/SGMEnergyTrajectory.cpp

```diff
@@ -0,0 +1,199 @@
+#include "SGMEnergyTrajectory.h"
+
+SGMEnergyTrajectory::SGMEnergyTrajectory(double startEnergy,
+                                         double endEnergy,
+                                         double time,
+                                         SGMGratingSupport::GratingTranslation gratingTranslation)
+{
+
+    startMonoInfo_ = new SGMMonochromatorInfo(startEnergy, gratingTranslation);
```

Comment:
There wasn't really a good state to initialize the mono info in for the case of the default constructor.  The calculations which follow (when setting the value of a given component, or requesting an energy for example) all tend to prompt a series of follow up calculations, and having the info initialized in an invalid state was causing a lot of corner cases in which these calculations failed (came out with NaN usually).

To get around this I made sure that there wasn't a default constructor for the info, and forced the setting of the values on initialization. This, however, means that I can't have one as a class member variable, without it being a pointer reference or using member initialization lists.